### PR TITLE
Clean up argument parsing.

### PR DESCRIPTION
**Pull request by evanlinjin**
Cleans up argument parsing and `cargo fmt` setup.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -205,41 +273,93 @@ async fn get_new_bech32_address(client: &mut tonic_lnd::Client) -> Address {
         .expect("lnd returned invalid address")
 }
 
-#[tokio::main]
-async fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let (config, mut args) = Config::including_optional_config_files(std::iter::empty::<&str>()).unwrap_or_exit();
+fn process_args<A: Iterator<Item = OsString>>(
```
Comment (DanGould) [inline src/main.rs:276]: separating this into a function makes way more sense. 讚

File: src/main.rs [hunk 2]
```diff
@@ -205,41 +273,93 @@ async fn get_new_bech32_address(client: &mut tonic_lnd::Client) -> Address {
...
+                        )
+                        .map_err(ArgError::InvalidWalletAmount),
+                    );
+                    None
+                }
+                unexpected_len => panic!("chunk len should only be 1 or 2, got {}", unexpected_len),
+            })
+            .collect::<Result<VecDeque<_>, _>>()?,
+        None => return Ok(None),
+    };
```
Comment (DanGould) [inline src/main.rs:318]: I like the comments that explain what is going on. However I think we should keep the, the original loop + match with `args.next()`. There's way less nesting and fewer lines. Can we merge the best of both and keep the comments but use the original logic?

**Comment by DanGould** [comment]
Let's merge each change as a separate commit.

Let's use the [rustfmt.toml](https://github.com/rust-bitcoin/rust-bitcoin/blob/master/rustfmt.toml) rules from the [long discussion about fmt rules in rust-bitcoin](https://github.com/rust-bitcoin/rust-bitcoin/pull/1040) to stay consistent with that community

**Review by DanGould — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 2 above)
Comment (evanlinjin) [reply src/main.rs]: @DanGould I tried again, this is my new solution. Let me know what you think.

```rust
    // parse scheduled channel arguments: pairs of (addr, amount)
    let mut channels = VecDeque::with_capacity(args.len() / 2);

    // the remaining single argument is the wallet amount in satoshis (if any)
    let wallet_amount = loop {
        match (args.next(), args.next()) {
            // we have a pair of arguments, interpret it as a scheduled channel (p2p addr, amount)
            (Some(addr), Some(amount)) => {
                channels.push_back(ScheduledChannel::from_args(addr, amount)?);
            }
            // if there is a remaining single argument, it is the wallet amount
            (Some(amount), None) => {
                break bitcoin::Amount::from_str_in(amount, bitcoin::Denomination::Satoshi)
                    .map_err(ArgError::InvalidWalletAmount)?
            }
            // if there is no remaining single argument, the wallet amount is 0
            _ => break bitcoin::Amount::ZERO,
        }
    };
```

Comment (DanGould) [reply src/main.rs]: beautiful!

File: src/main.rs [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+
+use args::ArgError;
 use bitcoin::util::address::Address;
 use bitcoin::util::psbt::PartiallySignedTransaction;
 use bitcoin::{Script, TxOut};
 use hyper::service::{make_service_fn, service_fn};
 use hyper::{Body, Method, Request, Response, Server, StatusCode};
 use ln_types::P2PAddress;
-use std::collections::HashMap;
+use std::collections::{HashMap};
```
Comment (DanGould) [inline src/main.rs:10]: rustfmt commit can fix this non-change

**Review by DanGould — APPROVED** [review]

**Pull request merged**